utils/crypto/bls/signer/rpcsigner/client.go

```diff
@@ -0,0 +1,94 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package rpcsigner
+
+import (
+	"context"
+	"errors"
+
+	"google.golang.org/grpc"
+
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+
+	pb "github.com/ava-labs/avalanchego/proto/pb/signer"
+)
+
+var (
+	_                   bls.Signer = (*Client)(nil)
+	ErrorEmptyPublicKey            = errors.New("empty public key")
+	ErrorEmptySignature            = errors.New("empty signature")
+)
+
+type Client struct {
+	client pb.SignerClient
+	conn   *grpc.ClientConn
+	pk     *bls.PublicKey
+}
+
+func NewClient(conn *grpc.ClientConn) (*Client, error) {
+	client := pb.NewSignerClient(conn)
+
+	pubkeyResponse, err := client.PublicKey(context.TODO(), &pb.PublicKeyRequest{})
+	if err != nil {
+		conn.Close()
```

Comment:
I feel weird about this code owning the lifecycle of the connection since it's provided by the caller. Some of these errors don't look [persistent](https://grpc.io/docs/guides/error/) + you can still re-use the connection since you can retry rpcs on different streams on the same http/2 connection even if this errored out.

You could probably make a strong case for some of these errors being persistent (i.e the empty public key case, the invalid pk case), but I would still think that you could handle all of these errors at the call-site through a connection close and have all the connection lifecycle managed there instead of having mixed ownership